src/SFA.DAS.Tasks.AcceptenceTests/Steps/Act.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Threading;
+using TechTalk.SpecFlow;
+using BoDi;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerAccounts.Events.Messages;
+using SFA.DAS.Tasks.AcceptenceTests.DependencyResolution;
+
+namespace SFA.DAS.Tasks.AcceptenceTests.Steps
+{
+    [Binding]
+    public class Act
+    {
+        private IObjectContainer _objectContainer;
+        private IAzureTopicMessageBus _azureTopicMessageBus;
+
+        public Act(IObjectContainer objectContainer)
+        {
+            _objectContainer = objectContainer;
+            _azureTopicMessageBus = _objectContainer.Resolve<IAzureTopicMessageBus>();
+        }
+
+        [When(@"(agreement_created|agreement_signed|legal_entity_removed) message get publish")]
+        public async Task WhenAgreement_CreatedMessageGetPublish(string message)
+        {
+            switch (message)
+            {
+                case "agreement_created":
+                    await PublishAndPeak<AgreementCreatedMessage>();
+                    break;
+                case "agreement_signed":
+                    await PublishAndPeak<AgreementSignedMessage>();
+                    break;
+                case "legal_entity_removed":
+                    await PublishAndPeak<LegalEntityRemovedMessage>();
+                    break;
+            }
+        }
+
+        private async Task PublishAndPeak<T>()
+        {
+            var agreement = _objectContainer.Resolve<T>();
+            await _azureTopicMessageBus.PublishAsync(agreement);
+            int count = 0;
+            while (true)
+            {
+                Thread.Sleep(5000);
```

Comment:
Its never normally good to have thread.sleeps in test as they hold up tests when potentially they don't need to be held up. We should really be mocking the last piece of code that gets the message and doing it that way so we can make the message available instantly but I'm aware this is asking too much at present to get these tests out the door. 

As a compromise you need to change 2 things:

1) change Thread.sleep to Task.Delay().Wait() as thread.sleep doesn't have to come back at that time and could potentially never come back. The CPU only has to take the thread back if it can after the sleep time has pasted. The Task.delay doesn't have this drawback.

2) reduce the wait time to something like 200 or 500 ms and increase the times allowed before failing. your could even reduce it to 50ms if you wanted to.